java/vector/src/test/java/org/apache/arrow/vector/TestListVector.java

```diff
@@ -810,15 +810,6 @@ public void testSetInitialCapacity() {
       vector.allocateNew();
       assertEquals(512, vector.getValueCapacity());
       assertEquals(8, vector.getDataVector().getValueCapacity());
-
-      boolean error = false;
-      try {
-        vector.setInitialCapacity(5, 0.1);
```

Comment:
Is it still valid to keep this to check that the capactiy is 1 (iiuc) and only remove the exception handling?

Reply:
No earlier we were throwing IllegalStateException but that shouldn't be done. Now we take a max and set it to 1.

Reply:
Yes, that I what I mean.  Still call `vector.setInitialCapacity(5, 0.1);` and just assert that capacity is 1 instead of trying to catch the exception

Reply:
you are right. added back the test with assertion.